index.js

```diff
@@ -136,8 +136,9 @@ function renderTag(elem, opts) {
   }
 
   if (
-    opts.xmlMode
-    && (!elem.children || elem.children.length === 0)
+    (opts.xmlMode
+    && (!elem.children || elem.children.length === 0)) ||
```

Comment:
I would prefer consistent behavior, as for `xmlMode`. When you're allowing self-closing tags in the first place, all tags should be treated this way. (There is no HTML/XML/XHTML version that allows self-closing tags solely for void elements.)